tests/topotests/bgp_maximum_prefix_invalid_update/test_bgp_maximum_prefix_invalid_update.py

```diff
@@ -83,29 +85,36 @@ def test_bgp_maximum_prefix_invalid():
     if tgen.routers_have_failure():
         pytest.skip(tgen.errors)
 
-    def _bgp_converge(router):
-        while True:
-            output = json.loads(
-                tgen.gears[router].vtysh_cmd("show ip bgp neighbor 192.168.255.1 json")
-            )
-            if output["192.168.255.1"]["connectionsEstablished"] > 0:
-                return True
+    r2 = tgen.gears["r2"]
 
-    def _bgp_parsing_nlri(router):
+    def _bgp_converge():
+        output = json.loads(r2.vtysh_cmd("show ip bgp neighbor 192.168.255.1 json"))
+        expected = {
+            "192.168.255.1": {
+                "connectionsEstablished": 1,
```

Comment:
I don't think this is safe. It could be possible that the connection flaps (due to load, for an example) and this counter would be different than expected.

Can we check for state `established` instead?

Reply:
The idea here is that the connection is closed immediately after it's established due to maximum-prefix limits. So, it SHOULD be always 1, not 2 or 3, or so.

Reply:
In that case wouldn't it be better to check the connection close motive?

E.g. `show bgp neighbor FOO json`:
```js
{
  "FOO": {
    "lastResetDueTo": "foo bar baz maximum prefix limit",
  }
}
```